walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(events.isEmpty() && request.getDuration() <= TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    
+    List<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    List<Event> eventList = new ArrayList<>(events);
```

Comment:
Your entire algorithm seems to rely on having the eventList sorted by start time. Is that guaranteed? If so you should make a comment about it

Reply:
The tests usually have sorted events, but the TimeRange class has a comparator which can sort them by start time. Should I just sort them in the beginning just to be sure?

Reply:
Tests can only check that your code has bugs, not the opposite. If the tests are written in a very specific way, it might still mean your code might have issues. So yes, you should sort it manually as there is no guarantee the input will be given to you already sorted.